flow/designs/gf180/uart-blocks/BLOCKS_grid_strategy.tcl

```diff
@@ -0,0 +1,35 @@
+####################################
+# global connections
+####################################
+add_global_connection -net {VDD} -inst_pattern {.*} -pin_pattern {^VDD$} -power
+add_global_connection -net {VDD} -inst_pattern {.*} -pin_pattern {^VDDPE$}
+add_global_connection -net {VDD} -inst_pattern {.*} -pin_pattern {^VDDCE$}
+add_global_connection -net {VDD} -inst_pattern {.*} -pin_pattern {^VDDP$}
+add_global_connection -net {VDD} -inst_pattern {.*} -pin_pattern {^VDDC$}
+add_global_connection -net {VDD} -inst_pattern {.*} -pin_pattern {^VNW$}
+add_global_connection -net {VSS} -inst_pattern {.*} -pin_pattern {^VSS$} -ground
+add_global_connection -net {VSS} -inst_pattern {.*} -pin_pattern {^VSSE$}
+add_global_connection -net {VSS} -inst_pattern {.*} -pin_pattern {^VSSC$}
+add_global_connection -net {VSS} -inst_pattern {.*} -pin_pattern {^VPW$}
+####################################
+# voltage domains
+####################################
+set_voltage_domain -name {CORE} -power {VDD} -ground {VSS}
+####################################
+# standard cell grid
+####################################
+define_pdn_grid -name {block} -voltage_domains {CORE} 
+add_pdn_stripe -grid {block} -layer {Metal1} -width {0.900} -pitch {5.040} -offset {0} -followpins
+add_pdn_stripe -grid {block} -layer {Metal4} -width {4.480} -spacing {0.56} -pitch {44.8} -offset {22.4}
+if { [info exists ::env(BLOCKS)] && [expr {$::env(BLOCKS) == $::env(DESIGN_NAME)}] } {
```

Comment:
@vvbandeira 
In `config.mk`, variable `BLOCKS` defined. But why I'm unable to read the variable in this tcl file?

Reply:
What makes you think the variable is not being read?

FYI, from a quick glance at the PR and its files, it does not look like the if expression will ever evaluate to true. There's only one place where you set `export BLOCKS = uart_rx` and no file where `DESIGN_NAME` is `uart_rx`.

Reply:
With BLOCKS variable uart_rx will be hardened before running uart-blocks.

During uart_rx run, i want if condition to run. For uart-blocks else to run to use all metal layers.

Is possible to implement?